pkg/build/strategies/sti/sti.go

```diff
@@ -169,15 +169,11 @@ func (b *STI) Prepare(request *api.Request) error {
 	}
 
 	// get the scripts
-	if request.ExternalRequiredScripts, err = b.installer.DownloadAndInstall(
-		b.requiredScripts, request.WorkingDir, true); err != nil {
+	if _, err := b.installer.InstallRequired(b.requiredScripts, request.WorkingDir); err != nil {
 		return err
 	}
-
-	if request.ExternalOptionalScripts, err = b.installer.DownloadAndInstall(
-		b.optionalScripts, request.WorkingDir, false); err != nil {
-		glog.Warningf("Failed downloading optional scripts: %v", err)
-	}
+	b.installer.InstallOptional(b.optionalScripts, request.WorkingDir)
```

Comment:
i would iterate over result and log the warnings about failed optional scripts download under V(2)